basics/timelock-contract/src/TimelockContract.cs

```diff
@@ -0,0 +1,106 @@
+using AElf.CSharp.Core.Extension;
+using AElf.Sdk.CSharp;
+using AElf.Types;
+using Google.Protobuf.WellKnownTypes;
+
+namespace AElf.Contracts.Timelock
+{
+    public partial class TimelockContract : TimelockContractContainer.TimelockContractBase
+    {
+        public override Empty Initialize(Empty input)
+        {
+            if (State.Initialized.Value)
+            {
+                return new Empty();
+            }
+            State.TokenContract.Value =
+                Context.GetContractAddressByName(SmartContractConstants.TokenContractSystemName);
+            State.Admin.Value = Context.Sender;
+            State.Initialized.Value = true;
+            return new Empty();
+        }
+        
+        public override Empty SetDelay(DelayInput input)
+        {
+            Assert(State.Admin.Value == Context.Sender, "No permission");
+            Assert(input != null, "Delay must not be null");
+            Assert(input.Delay >= TimelockContractConstants.MIN_DELAY, "Delay must exceed minimum delay");
+            Assert(input.Delay <= TimelockContractConstants.MAX_DELAY, "Delay must exceed maximum delay");
+            State.Delay.Value = (long) input.Delay;
+
+            Context.Fire(new NewDelay
+            {
+                NewDelay_ = input.Delay
+            });
+            return new Empty();
+        }
+
+        public override Empty ChangeAdmin(ChangeAdminInput input)
+        {
+            Assert(Context.Sender == Context.Self, "No permission");
+            Assert(input.NewAdmin != null, "NewAdmin must not be null");
+            State.Admin.Value = input.NewAdmin;
+            Context.Fire(new NewAdmin
+            {
+                NewAdmin_ = State.Admin.Value
+            });
+            return new Empty();
+        }
+
+        public override Empty QueueTransaction(TransactionInput input)
+        {
+            Assert(Context.Sender == State.Admin.Value, "No permission");
+            Assert(input.Eta >= Context.CurrentBlockTime.AddDays(State.Delay.Value), "Estimated execution block must satisfy delay");
```

Comment:
Does Ethereum code delay by number of days?

Reply:
Please change it to seconds. Solidity's 2 days = 2 * 24 * 60 * 60. The unit is second.

Reply:
Modified.